flink-core/src/main/java/org/apache/flink/configuration/DelegatingConfiguration.java

```diff
@@ -56,6 +57,7 @@ public DelegatingConfiguration() {
 	 */
 	public DelegatingConfiguration(Configuration backingConfig, String prefix)
 	{
+		Preconditions.checkNotNull(backingConfig);
 		this.backingConfig = backingConfig;
```

Comment:
Trivial note: We can also write `this.backingConfig = checkNotNull(backingConfig)` here.